main/src/ca/uwaterloo/flix/api/effectlock/EffectLock.scala

```diff
@@ -0,0 +1,46 @@
+package ca.uwaterloo.flix.api.effectlock
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.{RigidityEnv, Scheme, Type, TypeConstructor}
+import ca.uwaterloo.flix.language.ast.shared.Scope
+import ca.uwaterloo.flix.language.phase.unification.{EqualityEnv, Unification}
+
+object EffectLock {
+
+  /**
+    * Returns true if `sc1` is unifiable with `sc2` or if `sc1` is a monomorphic downgrade of `sc2`.
+    */
+  def isSafe(sc1: Scheme, sc2: Scheme)(implicit flix: Flix): Boolean = { // TODO: Check if new flix instance can be used locally
+    unifiableSchemes(sc1, sc2) || monomorphicDowngrade(sc1, sc2)
```

Comment:
@magnus-madsen I think I could use an extra check on this. Does this seem right?

Reply:
And if so, I am having some issues with checking monomorphic downgrade

Reply:
I think this is where @AStenbaek needs to join

Reply:
Its not entirely correct as it is written, since it allows bi-directional unification.

Reply:
Yes, that's also what the tests show